lambda/comprehend_pii_sync/app/sync_main.py

```diff
@@ -0,0 +1,159 @@
+        logger.error(f"InvalidToken for message: {message} ")
+    except step_functions_client.exceptions.TaskDoesNotExist:
+        logger.error(f"TaskDoesNotExist for message: {message} ")
+    except step_functions_client.exceptions.TaskTimedOut:
+        logger.error(f"TaskTimedOut for message: {message} ")
+
+
+def send_success_to_step_function(response, token, event, start_time):
+    logger.debug(f"comprehend result: {response}")
+
+    classification_result = "NONE"
+    for c in response['Entities']:
+        if c['Score'] > 0.50:
+            classification_result = c['Type']
+            break
+
+    call_duration = round(time.time() * 1000) - start_time
+    logger.info(
+        f"comprehend_sync_generic_call_duration_in_ms: {call_duration}")
+    try:
+        step_functions_client.send_task_success(
+            taskToken=token,
+            output=json.dumps({"documentType": classification_result}))
+    except step_functions_client.exceptions.InvalidToken:
+        logger.error(f"InvalidToken for message: {event} ")
+    except step_functions_client.exceptions.TaskDoesNotExist:
+        logger.error(f"TaskDoesNotExist for message: {event} ")
+    except step_functions_client.exceptions.TaskTimedOut:
+        logger.error(f"TaskTimedOut for message: {event} ")
+    except step_functions_client.exceptions.InvalidOutput:
+        logger.error(f"InvalidOutput for message: {event} ")
+
+
+def lambda_handler(event, _):
+    log_level = os.environ.get('LOG_LEVEL', 'INFO')
+    logger.setLevel(log_level)
+    logger.info(f"version: {version}")
+    logger.info(json.dumps(event))
+
+    language_code = os.environ.get('LANGUAGE_CODE', 'en')
+
+    processing_status: bool = True
+    files_with_failures: List[str] = list()
+
+    token = event['Token']
+    execution_id = event['ExecutionId']
+
+    if not "Payload" in event:
+        raise ValueError("Need Payload with manifest to process message.")
+
+    manifest: tm.IDPManifest = tm.IDPManifestSchema().load(event["Payload"]['manifest'])
+
+    s3_path = manifest.s3_path
+    payload = event['Payload']
+
+    if 'txt_output_location' in payload and "TextractOutputCSVPath" in payload[
+        'txt_output_location']:
+        s3_input_text = payload['txt_output_location']['TextractOutputCSVPath']
+    else:
+        raise ValueError(
+            f"no ['txt_output_location']['TextractOutputCSVPath'] to get the text file from "
+        )
+
+    logger.info(f"s3_path: {s3_path} \n \
+                    s3_input_text: {s3_input_text} \n\
+                token: {token} \n \
+                execution_id: {execution_id}")
+
+    try:
+        text = get_file_from_s3(s3_path=s3_input_text).decode('utf-8')[0:4900]
+
+        start_time = round(time.time() * 1000)
+        response = comprehend.detect_pii_entities(
+            Text=text,
+            LanguageCode=language_code,
+        )
+        send_success_to_step_function(response, token, event, start_time)
```

Comment:
All the blocks of textract + information about the offset.

Each words would have an offset.

Reply:
https://github.com/aws-samples/amazon-textract-idp-cdk-constructs/blob/3f40a57261d75c7a37d9a375472c816327fcad93/lambda/generatecsv/app/main.py#L96